### PR TITLE
refactor(gaussdb): move gaussdb resources to services path

### PR DESCRIPTION
**Pull request by Jason-Zhang9309**
**What this PR does / why we need it**:

move gaussdb resources to services path

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. move gaussdb resources to services path
2. fix the weak password problem in test cases
3. TestAccGaussDBProxy_basic failed but not caused by this commit, will fix it later
```

## PR Checklist

* [x] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/gaussdb' TESTARGS='-run TestAccGeminiDBDehResourceDataSource_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/gaussdb -v -run TestAccGeminiDBDehResourceDataSource_basic -timeout 360m -parallel 4
=== RUN   TestAccGeminiDBDehResourceDataSource_basic
=== PAUSE TestAccGeminiDBDehResourceDataSource_basic
=== CONT  TestAccGeminiDBDehResourceDataSource_basic
--- PASS: TestAccGeminiDBDehResourceDataSource_basic (11.15s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/gaussdb   11.238s

make testacc TEST='./huaweicloud/services/acceptance/gaussdb' TESTARGS='-run TestAccGeminiDBInstanceDataSource_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/gaussdb -v -run TestAccGeminiDBInstanceDataSource_basic -timeout 360m -parallel 4
=== RUN   TestAccGeminiDBInstanceDataSource_basic
=== PAUSE TestAccGeminiDBInstanceDataSource_basic
=== CONT  TestAccGeminiDBInstanceDataSource_basic
--- PASS: TestAccGeminiDBInstanceDataSource_basic (1103.77s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/gaussdb   1103.849s

make testacc TEST='./huaweicloud/services/acceptance/gaussdb' TESTARGS='-run TestAccGeminiDBInstancesDataSource_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/gaussdb -v -run TestAccGeminiDBInstancesDataSource_basic -timeout 360m -parallel 4
=== RUN   TestAccGeminiDBInstancesDataSource_basic
=== PAUSE TestAccGeminiDBInstancesDataSource_basic
=== CONT  TestAccGeminiDBInstancesDataSource_basic
--- PASS: TestAccGeminiDBInstancesDataSource_basic (1169.69s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/gaussdb   1169.844s

make testacc TEST='./huaweicloud/services/acceptance/gaussdb' TESTARGS='-run TestAccGaussdbMysqlConfigurationDataSource_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/gaussdb -v -run TestAccGaussdbMysqlConfigurationDataSource_basic -timeout 360m -parallel 4
=== RUN   TestAccGaussdbMysqlConfigurationDataSource_basic
=== PAUSE TestAccGaussdbMysqlConfigurationDataSource_basic
=== CONT  TestAccGaussdbMysqlConfigurationDataSource_basic
--- PASS: TestAccGaussdbMysqlConfigurationDataSource_basic (10.24s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/gaussdb   10.322s

make testacc TEST='./huaweicloud/services/acceptance/gaussdb' TESTARGS='-run TestAccGaussDBMysqlDehResourceDataSource_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/gaussdb -v -run TestAccGaussDBMysqlDehResourceDataSource_basic -timeout 360m -parallel 4
=== RUN   TestAccGaussDBMysqlDehResourceDataSource_basic
=== PAUSE TestAccGaussDBMysqlDehResourceDataSource_basic
=== CONT  TestAccGaussDBMysqlDehResourceDataSource_basic
--- PASS: TestAccGaussDBMysqlDehResourceDataSource_basic (8.78s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/gaussdb   8.884s

make testacc TEST='./huaweicloud/services/acceptance/gaussdb' TESTARGS='-run TestAccHuaweiCloudGaussdbMysqlFlavorsDataSource_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/gaussdb -v -run TestAccHuaweiCloudGaussdbMysqlFlavorsDataSource_basic -timeout 360m -parallel 4
=== RUN   TestAccHuaweiCloudGaussdbMysqlFlavorsDataSource_basic
=== PAUSE TestAccHuaweiCloudGaussdbMysqlFlavorsDataSource_basic
=== CONT  TestAccHuaweiCloudGaussdbMysqlFlavorsDataSource_basic
--- PASS: TestAccHuaweiCloudGaussdbMysqlFlavorsDataSource_basic (12.66s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/gaussdb   12.744s

make testacc TEST='./huaweicloud/services/acceptance/gaussdb' TESTARGS='-run TestAccGaussdbMysqlInstanceDataSource_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/gaussdb -v -run TestAccGaussdbMysqlInstanceDataSource_basic -timeout 360m -parallel 4
=== RUN   TestAccGaussdbMysqlInstanceDataSource_basic
=== PAUSE TestAccGaussdbMysqlInstanceDataSource_basic
=== CONT  TestAccGaussdbMysqlInstanceDataSource_basic
--- PASS: TestAccGaussdbMysqlInstanceDataSource_basic (1242.26s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/gaussdb   1242.342s

make testacc TEST='./huaweicloud/services/acceptance/gaussdb' TESTARGS='-run TestAccGaussdbMysqlInstancesDataSource_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/gaussdb -v -run TestAccGaussdbMysqlInstancesDataSource_basic -timeout 360m -parallel 4
=== RUN   TestAccGaussdbMysqlInstancesDataSource_basic
=== PAUSE TestAccGaussdbMysqlInstancesDataSource_basic
=== CONT  TestAccGaussdbMysqlInstancesDataSource_basic
--- PASS: TestAccGaussdbMysqlInstancesDataSource_basic (1222.31s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/gaussdb   1222.392s

make testacc TEST='./huaweicloud/services/acceptance/gaussdb' TESTARGS='-run TestAccGaussDBNoSQLFlavors_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/gaussdb -v -run TestAccGaussDBNoSQLFlavors_basic -timeout 360m -parallel 4
=== RUN   TestAccGaussDBNoSQLFlavors_basic
=== PAUSE TestAccGaussDBNoSQLFlavors_basic
=== CONT  TestAccGaussDBNoSQLFlavors_basic
--- PASS: TestAccGaussDBNoSQLFlavors_basic (22.74s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/gaussdb   22.853s

make testacc TEST='./huaweicloud/services/acceptance/gaussdb' TESTARGS='-run TestAccGaussRedisInstanceDataSource_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/gaussdb -v -run TestAccGaussRedisInstanceDataSource_basic -timeout 360m -parallel 4
=== RUN   TestAccGaussRedisInstanceDataSource_basic
=== PAUSE TestAccGaussRedisInstanceDataSource_basic
=== CONT  TestAccGaussRedisInstanceDataSource_basic
--- PASS: TestAccGaussRedisInstanceDataSource_basic (1203.02s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/gaussdb   1203.103s

make testacc TEST='./huaweicloud/services/acceptance/gaussdb' TESTARGS='-run TestAccGeminiDBInstance_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/gaussdb -v -run TestAccGeminiDBInstance_basic -timeout 360m -parallel 4
=== RUN   TestAccGeminiDBInstance_basic
=== PAUSE TestAccGeminiDBInstance_basic
=== CONT  TestAccGeminiDBInstance_basic
--- PASS: TestAccGeminiDBInstance_basic (1104.60s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/gaussdb   1104.686s

make testacc TEST='./huaweicloud/services/acceptance/gaussdb' TESTARGS='-run TestAccGaussDBInstance_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/gaussdb -v -run TestAccGaussDBInstance_basic -timeout 360m -parallel 4
=== RUN   TestAccGaussDBInstance_basic
=== PAUSE TestAccGaussDBInstance_basic
=== CONT  TestAccGaussDBInstance_basic
--- PASS: TestAccGaussDBInstance_basic (1318.64s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/gaussdb   1318.718s

make testacc TEST='./huaweicloud/services/acceptance/gaussdb' TESTARGS='-run TestAccGaussRedisInstance_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/gaussdb -v -run TestAccGaussRedisInstance_basic -timeout 360m -parallel 4
=== RUN   TestAccGaussRedisInstance_basic
=== PAUSE TestAccGaussRedisInstance_basic
=== CONT  TestAccGaussRedisInstance_basic
--- PASS: TestAccGaussRedisInstance_basic (3437.18s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/gaussdb   3437.275s

make testacc TEST='./huaweicloud/services/acceptance/gaussdb' TESTARGS='-run TestAccOpenGaussInstanceDataSource_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/gaussdb -v -run TestAccOpenGaussInstanceDataSource_basic -timeout 360m -parallel 4
=== RUN   TestAccOpenGaussInstanceDataSource_basic
=== PAUSE TestAccOpenGaussInstanceDataSource_basic
=== CONT  TestAccOpenGaussInstanceDataSource_basic
--- PASS: TestAccOpenGaussInstanceDataSource_basic (1760.20s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/gaussdb   1760.282s

make testacc TEST='./huaweicloud/services/acceptance/gaussdb' TESTARGS='-run TestAccOpenGaussInstance_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/gaussdb -v -run TestAccOpenGaussInstance_basic -timeout 360m -parallel 4
=== RUN   TestAccOpenGaussInstance_basic
=== PAUSE TestAccOpenGaussInstance_basic
=== CONT  TestAccOpenGaussInstance_basic
--- PASS: TestAccOpenGaussInstance_basic (1752.83s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/gaussdb   1752.917s

make testacc TEST='./huaweicloud/services/acceptance/gaussdb' TESTARGS='-run TestAccOpenGaussInstancesDataSource_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/gaussdb -v -run TestAccOpenGaussInstancesDataSource_basic -timeout 360m -parallel 4
=== RUN   TestAccOpenGaussInstancesDataSource_basic
=== PAUSE TestAccOpenGaussInstancesDataSource_basic
=== CONT  TestAccOpenGaussInstancesDataSource_basic
--- PASS: TestAccOpenGaussInstancesDataSource_basic (3521.07s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/gaussdb   3521.212s

make testacc TEST='./huaweicloud/services/acceptance/gaussdb' TESTARGS='-run TestAccGaussDBProxy_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/gaussdb -v -run TestAccGaussDBProxy_basic -timeout 360m -parallel 4
=== RUN   TestAccGaussDBProxy_basic
=== PAUSE TestAccGaussDBProxy_basic
=== CONT  TestAccGaussDBProxy_basic
    resource_huaweicloud_gaussdb_mysql_proxy_test.go:35: Step 1/2 error: Check failed: Check 2/3 error: huaweicloud_gaussdb_mysql_proxy.test: Attribute 'flavor' expected "gaussdb.proxy.xlarge.arm.2", got ""
--- FAIL: TestAccGaussDBProxy_basic (1551.97s)
FAIL
FAIL    github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/gaussdb   1552.067s
FAIL
GNUmakefile:21: recipe for target 'testacc' failed
make: *** [testacc] Error 1
```

### CODE REVIEW

File: huaweicloud/services/gaussdb/resource_huaweicloud_gaussdb_opengauss_instance.go [hunk 1]
```diff
@@ -430,16 +431,16 @@ func resourceOpenGaussInstanceCreate(d *schema.ResourceData, meta interface{}) e
 
 func resourceOpenGaussInstanceRead(d *schema.ResourceData, meta interface{}) error {
 	config := meta.(*config.Config)
-	region := GetRegion(d, config)
-	client, err := config.OpenGaussV3Client(GetRegion(d, config))
+	region := config.GetRegion(d)
+	client, err := config.OpenGaussV3Client(config.GetRegion(d))
```
Comment (Lance52259) [inline huaweicloud/services/gaussdb/resource_huaweicloud_gaussdb_opengauss_instance.go]: s/config.OpenGaussV3Client(GetRegion(d, config))/config.OpenGaussV3Client(region)

Comment (Jason-Zhang9309) [reply huaweicloud/services/gaussdb/resource_huaweicloud_gaussdb_opengauss_instance.go]: fixed

**Pull request merged**